src/main/java/teammates/common/datatransfer/FeedbackConstantSumQuestionDetails.java

```diff
@@ -397,16 +398,46 @@ public String getQuestionResultStatisticsHtml(
         Map<String, List<Integer>> optionPoints = generateOptionPointsMapping(responses);
 
         DecimalFormat df = new DecimalFormat("#.##");
+        /*
+        Map<String, List<Integer>> sortedOptionPoints = new TreeMap<String, List<Integer>>(new Comparator<String>() {
+            @Override
+            public int compare(String s1, String s2) {
+                return s2.compareTo(s1);
+            }
+        });
+        */
+        Map<String, List<Integer>> sortedOptionPoints = new TreeMap<String, List<Integer>>();
+        
+        Map<String, String> identifierMap = new HashMap<String, String>();
         
         for (Entry<String, List<Integer>> entry : optionPoints.entrySet()) {
             
+            if (distributeToRecipients) {
+                
+                String participantIdentifier = entry.getKey();
+                String name = bundle.getNameForEmail(participantIdentifier);
+                
+                identifierMap.put(name, participantIdentifier);
+                sortedOptionPoints.put(name, entry.getValue());
+                
+            } else {
+                
+                String option = options.get(Integer.parseInt(entry.getKey()));
+                
+                identifierMap.put(option, entry.getKey());
+                sortedOptionPoints.put(option, entry.getValue());
```

Comment:
Again, this would result in problems if there are 2 or more same options. However this is an issue of its own, to be fixed in another PR, as it does not make sense to have 2 or more same options. Users should be prevented from setting questions with 2 or more same options.